ipatests/test_integration/test_installation.py

```diff
@@ -404,6 +404,46 @@ def test_install_dns(self):
             extra_args=['--dnssec-master', '--no-dnssec-validation']
         )
 
+    def test_ipactl_restart_pki_tomcat(self):
+        """ Test if ipactl restart restarts the pki-tomcatd
+
+        Wrong logic was triggering the start instead of restart
+        for pki-tomcatd. This test validates that restart
+        called on pki-tomcat properly.
+
+        related ticket : https://pagure.io/freeipa/issue/7927
+        """
+        splitted = []
+        # get process id of pki-tomcatd
+        cmd = self.master.run_command(['ps', '-ef'])
+        for line in cmd.stdout_text.split('\n'):
+            if "pki-tomcatd" in line:
+                splitted = line.split()
+        pki_pid = splitted[1]
+
+        # check if pki-tomcad restarted
+        cmd = self.master.run_command(['ipactl', 'restart'])
+        assert "Restarting pki-tomcatd Service" in cmd.stdout_text
+
+        # check if pid for pki-tomcad changed
+        cmd = self.master.run_command(['ps', '-ef'])
+        for line in cmd.stdout_text.split('\n'):
+            if "pki-tomcatd" in line:
+                splitted = line.split()
+        assert pki_pid not in splitted
+        pki_pid = splitted[1]
+
+        # check if pki-tomcad restarted
+        cmd = self.master.run_command(['ipactl', 'restart'])
+        assert "Restarting pki-tomcatd Service" in cmd.stdout_text
+
+        # check if pid for pki-tomcad changed
+        cmd = self.master.run_command(['ps', '-ef'])
+        for line in cmd.stdout_text.split('\n'):
+            if "pki-tomcatd" in line:
+                splitted = line.split()
+        assert pki_pid not in splitted
```

Comment:
Rather than duplicating this same block three times it would be better in a separate function.